test/plugin/base_test.rb

```diff
@@ -607,32 +607,44 @@ def test_sanitize_tags_from_container_name_with_require_valid_tags_false
 
   def test_timestamps
     setup_gce_metadata_stubs
-    [Time.at(123_456.789), Time.at(0), Time.now].each do |ts|
+    current_time = Time.now
+    next_year = Time.mktime(current_time.year + 1)
+    one_second_before_next_year = next_year - 1
+    adjusted_to_last_year =
+      one_second_before_next_year.to_datetime.prev_year.to_time
+    one_day_into_next_year = next_year.to_date.next_day.to_time
+    {
+      Time.at(123_456.789) => Time.at(123_456.789),
+      Time.at(0) => Time.at(0),
+      current_time => current_time,
+      one_second_before_next_year => adjusted_to_last_year,
+      one_day_into_next_year => Time.at(0)
```

Comment:
Also `one_second_into_the_next_year` and `next_year`?

Reply:
Added.